bitcoin/src/crypto/key.rs

```diff
@@ -928,17 +928,24 @@ impl From<secp256k1::Error> for FromSliceError {
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[non_exhaustive]
 pub enum FromWifError {
-    /// A base58 error.
+    /// A base58 decoding error.
     Base58(base58::Error),
+    /// Base58 decoded data was an invalid length.
+    InvalidBase58PayloadLength(InvalidBase58PayloadLengthError),
```

Comment:
It'd be nicer to just call this `InvalidLength` and present the number of chars to the user.

Reply:
I can get behind that change though!

Reply:
I had a bit of a play with doing this but because one error uses expected length and one uses expected length I believe its more clear if left as is.

Reply:
I don't understand, can you explain more?

Reply:
Wow, that was bad English I wrote. Should have been - one uses expected length and one uses "should have been 33 or 34". WIP accepts data length of 33 or 34

Reply:
We could distinguish those internally with an enum but maybe it's too much. I suggest we at least make the error messages more user-friendly.

Reply:
I looked at the error messages and did not see exactly where you wanted them improved. FTR I'm not confident that all our nested errors stack error messages in a meaningful, non-duplicate way, I've kind of given up for now and am just putting something grep'able. I think we will need to come back pre-1.0 and do some deep investigation like I tried to do in https://github.com/rust-bitcoin/rust-bech32/pull/151